File: backend/src/main/java/com/example/tracker/service/MaquinaSoftwareInstaladoService.java
package com.example.tracker.service;

import com.example.tracker.dto.maquinasoftwareinstalado.MaquinaSoftwareInstaladoCreateDTO;
import com.example.tracker.entity.MaquinaSoftwareInstalado;
import java.util.List;

public interface MaquinaSoftwareInstaladoService {

    List<MaquinaSoftwareInstalado> listarTodos();

    MaquinaSoftwareInstalado buscarPorId(Integer id);

    List<MaquinaSoftwareInstalado> buscarPorMaquinaContrato(Integer codigoMaquinaContrato);

    List<MaquinaSoftwareInstalado> buscarPorSoftware(Integer codigoSoftware);

    MaquinaSoftwareInstalado cadastrar(MaquinaSoftwareInstaladoCreateDTO dto);

    MaquinaSoftwareInstalado atualizar(Integer id, MaquinaSoftwareInstaladoCreateDTO dto);

    void deletar(Integer id);
}

File: frontend/src/services/maquinaSoftwareInstaladoService.ts
import { apiFetch } from './api';

export interface MaquinaSoftwareInstaladoResponseDTO {
  codigo: number;
  codigoMaquinaContrato: number;
  codigoSoftware: number;
  versaoInstalada: string;
  chaveLicenca?: string;
}

export interface MaquinaSoftwareInstaladoCreateDTO {
  codigoMaquinaContrato: number;
  codigoSoftware: number;
  versaoInstalada: string;
  chaveLicenca?: string;
}

export const maquinaSoftwareInstaladoService = {
  listar: () =>
    apiFetch<MaquinaSoftwareInstaladoResponseDTO[]>('/maquinas-softwares-instalados'),

  buscarPorMaquinaContrato: (codigoMaquinaContrato: number) =>
    apiFetch<MaquinaSoftwareInstaladoResponseDTO[]>(
      `/maquinas-softwares-instalados?codigoMaquinaContrato=${codigoMaquinaContrato}`
    ),

  buscarPorSoftware: (codigoSoftware: number) =>
    apiFetch<MaquinaSoftwareInstaladoResponseDTO[]>(
      `/maquinas-softwares-instalados?codigoSoftware=${codigoSoftware}`
    ),

  buscarPorId: (id: number) =>
    apiFetch<MaquinaSoftwareInstaladoResponseDTO>(`/maquinas-softwares-instalados/${id}`),

  criar: (data: MaquinaSoftwareInstaladoCreateDTO) =>
    apiFetch<MaquinaSoftwareInstaladoResponseDTO>('/maquinas-softwares-instalados', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  atualizar: (id: number, data: MaquinaSoftwareInstaladoCreateDTO) =>
    apiFetch<MaquinaSoftwareInstaladoResponseDTO>(`/maquinas-softwares-instalados/${id}`, {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  deletar: (id: number) =>
    apiFetch<void>(`/maquinas-softwares-instalados/${id}`, {
      method: 'DELETE',
    }),
};